### PR TITLE
feat(ingest/dagster): Dagster asset dep capture

### PR DESCRIPTION
**Pull request by treff7es**
## Checklist

- [ ] The PR conforms to DataHub's [Contributing Guideline](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable). If a new feature has been added a Usage Guide has been added for the same.
- [ ] For any breaking change/potential downtime/deprecation/big changes an entry has been made in [Updating DataHub](https://github.com/datahub-project/datahub/blob/master/docs/how/updating-datahub.md)

## Summary by CodeRabbit

- **New Features**
	- Enhanced documentation for the DataHub integration with Dagster, improving readability and clarity.
	- New parameters and methods added to the datahub sensors for improved job monitoring and asset metadata handling.
	- Introduction of a new `Lineage` class to track asset relationships more effectively.

- **Improvements**
	- Streamlined installation instructions and prerequisites for the DataHub Dagster plugin.
	- Expanded configuration options for asset management with detailed descriptions.

- **Bug Fixes**
	- Refactored asset materialization processing to include enhanced logging and metadata handling.

### CODE REVIEW

File: metadata-ingestion-modules/dagster-plugin/src/datahub_dagster_plugin/client/dagster_generator.py [hunk 1]
```diff
@@ -495,10 +739,92 @@ def emit_asset(
...
+        if downstreams:
+            for downstream in downstreams:
+                if materialize_dependencies:
+                    mcp = MetadataChangeProposalWrapper(
+                        entityUrn=downstream,
+                        aspect=DatasetKeyClass(
+                            name=DatasetUrn.create_from_string(downstream).name,
+                            platform=DatasetUrn.create_from_string(downstream).platform,
+                            origin=DatasetUrn.create_from_string(downstream).env,
+                        ),
```
Comment (hsheth2) [inline metadata-ingestion-modules/dagster-plugin/src/datahub_dagster_plugin/client/dagster_generator.py:765]: use `DatasetUrn(...).to_key_aspect()`

File: metadata-ingestion-modules/dagster-plugin/src/datahub_dagster_plugin/client/dagster_generator.py [hunk 2]
```diff
@@ -495,10 +739,92 @@ def emit_asset(
...
+                    platform=make_data_platform_urn(DAGSTER_PLATFORM),
                 ),
             )
             graph.emit_mcp(mcp)
+
+        tag_mcp = self.generate_asset_group_tag(
+            graph, asset_urn=dataset_urn, target_urn=dataset_urn
+        )
+        if tag_mcp:
+            print(f"tag_mcp: {tag_mcp}")
```
Comment (hsheth2) [inline metadata-ingestion-modules/dagster-plugin/src/datahub_dagster_plugin/client/dagster_generator.py:753]: logger

**Review by hsheth2 — COMMENTED** [review]
Gave this a quick skim - haven't reviewed it in depth yet

**Review by jjoyce0510 — APPROVED** [review]
Can we please get this merged? It's been sitting since may

**Comment by jjoyce0510** [comment]
Let's please get this in! It has great improvements on both code and docs!

**Pull request closed without merging**